vyos/vyos-1x
# OpenVPN: T3350: Changed custom options for OpenVPN processing

## Change Summary

Changed custom options for OpenVPN processing

## Types of changes


## Related Task(s)

* https://phabricator.vyos.net/T3350

## Component(s) name

openvpn

## Proposed changes

Custom OpenVPN options moved back to the command line from a configuration file. This should keep full compatibility with the
`crux` branch, and allows to avoid mistakes with parsing options that contain `--` in the middle. The only smart part of this - handling a `push` option. Because of internal changes in OpenVPN, previously it did not require an argument in the double-quotes, but after version update in `equuleus` and `sagitta` old syntax became invalid. So, all the `push` options are processed to add quotes. The solution is still not complete, because if a single config line contains `push` with other options, it will not work, but it is better than nothing.

## How to test

Create OpenVPN interface with options:
```
openvpn-option "plugin /config/openvpn_cc_script.so &quot;/config/scripts/script.sh --cache-dir /tmp/cache check --gateway \\&quot;Gateway 1\\&quot; /dev/null&quot;"
openvpn-option '--push dhcp-option DNS 1.1.1.1'
```
Before the changes in this commit, OpenVPN interface will not be able to start because of broken config syntax. After the commit, the OpenVPN process should start with options in a command line.

## Checklist:
